parcel_test.go

```diff
@@ -28,94 +33,233 @@ func getTestParcel() Parcel {
+
+	// Возврат подключенной базы данных
+	return db
+}
+
+// TestAddGetDelete - тест для проверки операций создания, получения и удаления посылки
 func TestAddGetDelete(t *testing.T) {
-	// prepare
-	db, err := // настройте подключение к БД
+	// Подготовка окружения и автоматическое закрытие БД после теста
+	db := setupDatabase(t)
+	defer db.Close()
+
+	// Создание хранилища посылок и получение тестовой посылки
 	store := NewParcelStore(db)
 	parcel := getTestParcel()
+	var err error
 
-	// add
-	// добавьте новую посылку в БД, убедитесь в отсутствии ошибки и наличии идентификатора
+	// Структура для хранения тестовых кейсов
+	tests := []struct {
+		name     string                                // Название тестового кейса
+		testFunc func(*testing.T, ParcelStore, Parcel) // Функция, реализующая логику теста
+	}{
+		{
+			name: "Parcel insertion test",
+			testFunc: func(*testing.T, ParcelStore, Parcel) {
+				parcel.Number, err = store.Add(parcel)
+				assert.NotEmpty(t, parcel.Number, "parcel ID should not be empty after insertion. Test parcel: %v", parcel)
+				require.NoError(t, err, "failed to insert parcel into database. Parcel details: %v. Error: %v", parcel, err)
+			},
+		},
+		{
+			name: "Parcel retrieval test by ID",
+			testFunc: func(*testing.T, ParcelStore, Parcel) {
+				res, err := store.Get(parcel.Number)
+				require.NoError(t, err, "failed to retrieve parcel with ID %d from database. Error: %v", parcel.Number, err)
 
-	// get
-	// получите только что добавленную посылку, убедитесь в отсутствии ошибки
-	// проверьте, что значения всех полей в полученном объекте совпадают со значениями полей в переменной parcel
+				assert.Equal(t, res.Address, parcel.Address, "address mismatch. Expected: %v, Actual: %v", parcel.Address, res.Address)
+				assert.Equal(t, res.Client, parcel.Client, "client mismatch. Expected: %v, Actual: %v", parcel.Client, res.Client)
+				assert.Equal(t, res.CreatedAt, parcel.CreatedAt, "createdAt mismatch. Expected: %v, Actual: %v", parcel.CreatedAt, res.CreatedAt)
+				assert.Equal(t, res.Number, parcel.Number, "ID mismatch. Expected: %v, Actual: %v", parcel.Number, res.Number)
+				assert.Equal(t, res.Status, parcel.Status, "status mismatch. Expected: %v, Actual: %v", parcel.Status, res.Status)
+			},
+		},
+		{
+			name: "Parcel deletion test",
+			testFunc: func(*testing.T, ParcelStore, Parcel) {
+				err = store.Delete(parcel.Number)
+				require.NoError(t, err, "failed to delete parcel with ID %d from database", parcel.Number)
 
-	// delete
-	// удалите добавленную посылку, убедитесь в отсутствии ошибки
-	// проверьте, что посылку больше нельзя получить из БД
+				_, err = store.Get(parcel.Number)
+				require.Error(t, err, "expected error when trying to retrieve deleted parcel with ID %d", parcel.Number)
+
+				originalErr := errors.Unwrap(err)
+				require.Equal(t, sql.ErrNoRows, originalErr, "expected specific sql.ErrNoRows error when searching for deleted parcel with ID %d", parcel.Number)
```

Comment:
Для сравнивания ошибок нужно использовать специализированную функцию
```suggestion
	require.ErrorIs(t, originalErr, sql.ErrNoRows, "expected specific sql.ErrNoRows error when searching for deleted parcel with ID %d", parcel.Number)
```